### PR TITLE
Kernel config wibbling

### PR DESCRIPTION
**Pull request by K900**
###### Description of changes

Enable DAMON, Zen3+ branch sampling, Bay Trail stuff. DAMON is the big one, it doesn't do anything by default, see [here](https://sjp38.github.io/post/damon_evaluation/) for how to enable/test it.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by RaitoBezarius — APPROVED** [review]
LGTM; untested

File: pkgs/os-specific/linux/kernel/common-config.nix [hunk 1]
```diff
@@ -617,6 +642,11 @@ let
       RING_BUFFER_BENCHMARK = no;
     };
 
+    perf = {
+      # enable AMD Zen branch sampling if available
+      PERF_EVENTS_AMD_BRS       = (option yes);
```
Comment (alyssais) [inline pkgs/os-specific/linux/kernel/common-config.nix:647]:
```suggestion
      PERF_EVENTS_AMD_BRS       = option yes;
```

**Review by alyssais — APPROVED** [review]

File: pkgs/os-specific/linux/kernel/common-config.nix (hunk 1 above)
Comment (K900) [reply pkgs/os-specific/linux/kernel/common-config.nix]: There.

**Comment by vcunat** [comment]
This merge broke config of some older kernels.  I tried the merge commit and its parent to confirm (on 5.10).  List: https://hydra.nixos.org/eval/1796896?filter=.kernel.&compare=1796824

**Comment by K900** [comment]
That's strange. The option should be there in 5.10 according to LKDDB: https://cateee.net/lkddb/web-lkddb/TPS68470_PMIC_OPREGION.html

**Comment by vcunat** [comment]
Maybe some details changed, like their prerequisites?

**Comment by K900** [comment]
I'll have to double check tomorrow.

**Comment by vcunat** [comment]
nixos-unstable channel will probably get updated overnight (after almost a week), including this breakage.  I don't see an easy way to stop it now, as the `tested` job completed some hours ago already.

**Comment by K900** [comment]
We don't need to rebuild all the kernels to fix this, so it should be fine I think.

**Pull request merged**